streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -167,18 +167,7 @@ void handleCorruption(final Set<TaskId> corruptedTasks) {
             }
         }
 
-        // Make sure to clean up any corrupted standby tasks in their entirety before committing
-        // since TaskMigrated can be thrown and the resulting handleLostAll will only clean up active tasks
         closeAndRevive(corruptedStandbyTasks);
-
-        commit(tasks()
```

Comment:
@guozhangwang I traced all the way back to where this TaskCorrupted logic was initially added in [#8058](https://github.com/apache/kafka/pull/8058) and re-read the relevant discussions, and AFAICT there was never an explicit reason for committing. I guess we just never questioned it since the beginning.

(In fact it seems to have only ever caused problems: eg [KAFKA-10144](https://issues.apache.org/jira/browse/KAFKA-10144))

Reply:
I dug a bit further, and I think the reason for this commit is here: https://github.com/apache/kafka/pull/8440/files#r407722022

This is for EOS-beta specifically: inside handleCorrupted we would call `task.closeDirty` and which would call `recordCollector.closeDirty`, which would call `producer.abortTxn`. In EOS-beta, this would abort transaction for all tasks.

So I feel maybe we should refactor this a bit, i.e. in `closeAndRevive` we do not close its record collector since we know the task is going to be revived anyways, but instead we need to cleanup the collector (e.g. reset exception flag, clear sent future bookkeeping etc), to workaround it. In this way we would not need to commit before `handleCorrupted`. WDYT?

Also cc @mjsax who worked on #8440

Reply:
I am fine with refactoring. On the other hand, does it add even more "corner" case and make the code harder to reason about?

I "simpler" solution might be to just risk that the commit fails (I think be risk is low), and if it indeed fails, we throw a `TaskMigratedException` that is already handled in the code.

Reply:
I think we do need to do the `commit` including the `abortTxn` since we can now get a TaskCorrupted at any point, not just on RESTORING tasks which don't have anything to abort. But since we're already in `handleCorrupted` we may as well just catch the TaskCorrupted, tack on any new corrupted tasks, and proceed with the closeAndRevive